packages/dataviews/src/view-grid.js

```diff
@@ -135,6 +131,49 @@ export default function ViewGrid( {
 	selection,
 	onSelectionChange,
 } ) {
+	const gridRef = useRef( null );
+	const eventControllerRef = useRef( null );
+	const [ hasNoPointerEvents, setHasNoPointerEvents ] = useState( false );
+	const [ pointerIsWithinBounds, setPointerIsWithinBounds ] =
+		useState( false );
+
+	const pointerIsWithinBoundsHandler = useCallback(
+		( { metaKey, ctrlKey, type } ) => {
+			const isWithinBounds = type === 'mouseenter';
+			const isKeyPressed = isAppleOS() ? metaKey : ctrlKey;
+			setPointerIsWithinBounds( isWithinBounds );
+			setHasNoPointerEvents( isWithinBounds && isKeyPressed );
+		},
+		[]
+	);
+
+	useEffect( () => {
+		if ( eventControllerRef.current ) {
+			eventControllerRef.current.abort();
+		}
+
+		const doc = gridRef.current?.ownerDocument;
+		if ( ! doc ) return;
+
+		if ( pointerIsWithinBounds ) {
+			const listener = ( { key, type } ) => {
+				if ( key === ( isAppleOS() ? 'Meta' : 'Control' ) ) {
+					setHasNoPointerEvents( type === 'keydown' );
+				}
+			};
+			const controller = new AbortController();
+			controller.signal.addEventListener( 'abort', () => {
+				eventControllerRef.current = null;
+			} );
+			const config = { capture: true, signal: controller.signal };
+			doc.body.addEventListener( 'keydown', listener, config );
+			doc.body.addEventListener( 'keyup', listener, config );
```

Comment:
Adding `keyup`/`keydown` listeners when the mouse pointer enters the grid, and removing when the mouse pointer leaves again. The event listeners are added to the body so that the key press is captured regardless of which element has focus.